helpdesk_mgmt_stage_server_action/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+        for record in records:
+            action = record.stage_id.action_id
+            if action:
+                context = {
+                    "active_model": self._name,
+                    "active_ids": [record.id],
+                }
+                action.with_context(**context).run()
+        return records
+
+    def write(self, vals):
+        res = super().write(vals)
+        action = (
+            self.env["helpdesk.ticket.stage"].browse(vals["stage_id"]).action_id
+            if "stage_id" in vals
+            else None
+        )
+        if action:
```

Comment:
We don't want that, because here, you execute the action each time we have `stage_id` in `vals`.
But if you call write function with the current `stage_id`, you will execute the action even if finally, the stage didn't changed.
Then please execute the action only if the stage really change.